From bf31d8195f13402b0a272c7bfe52ecb84faad7fe Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 15:30:28 -0400
Subject: [ethernet] igbvf: Use pci_enable_msix_range() instead of pci_enable_msix()

Message-id: <1409067034-17415-8-git-send-email-sassmann@redhat.com>
Patchwork-id: 89220
O-Subject: [RHEL7.1 PATCH 07/13] igbvf: Use pci_enable_msix_range() instead of pci_enable_msix()
Bugzilla: 1091122
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

As result of deprecation of MSI-X/MSI enablement functions
pci_enable_msix() and pci_enable_msi_block() all drivers
using these two interfaces need to be updated to use the
new pci_enable_msi_range() and pci_enable_msix_range()
interfaces.

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Cc: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Cc: Bruce Allan <bruce.w.allan@intel.com>
Cc: e1000-devel@lists.sourceforge.net
Cc: netdev@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 4601e7591f2609860b432206cd65845615c8b45a)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igbvf/netdev.c b/drivers/net/ethernet/intel/igbvf/netdev.c
index 675435f..e2c6d80 100644
--- a/drivers/net/ethernet/intel/igbvf/netdev.c
+++ b/drivers/net/ethernet/intel/igbvf/netdev.c
@@ -1043,11 +1043,11 @@ static void igbvf_set_interrupt_capability(struct igbvf_adapter *adapter)
   for (i = 0; i < 3; i++)
    adapter->msix_entries[i].entry = i;
 
-  err = pci_enable_msix(adapter->pdev,
-                        adapter->msix_entries, 3);
+  err = pci_enable_msix_range(adapter->pdev,
+                              adapter->msix_entries, 3, 3);
  }
 
- if (err) {
+ if (err < 0) {
   /* MSI-X failed */
   dev_err(&adapter->pdev->dev,
           "Failed to initialize MSI-X interrupts.\n");
-- 
1.7.1